### PR TITLE
Add ByteArray

### PR DESCRIPTION
**Pull request by DelevoXDG**
## Describe your changes

Related to and is needed for #428 

- Add `StarknetTypedData` that corresponds to `ByteArray` struct in Cairo

## Linked issues

Closes

## Breaking changes

- [ ] This issue contains breaking changes

## References

- [Serialization of byte arrays](https://docs.starknet.io/documentation/architecture_and_concepts/Smart_Contracts/serialization_of_Cairo_types/#serialization_of_byte_arrays)
- Tests in [cairo repo](https://github.com/starkware-libs/cairo/blob/b22e43ffecf884877ed3a65d3077c0b8af586d6c/corelib/src/test/byte_array_test.cairo#L458)

### CODE REVIEW

File: lib/src/main/kotlin/com/swmansion/starknet/data/types/StarknetByteArray.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package com.swmansion.starknet.data.types
+
+import com.swmansion.starknet.data.types.conversions.ConvertibleToCalldata
+import com.swmansion.starknet.extensions.splitToShortStrings
+import com.swmansion.starknet.extensions.toFelt
+
+data class StarknetByteArray(
```
Comment (DelevoXDG) [inline lib/src/main/kotlin/com/swmansion/starknet/data/types/StarknetByteArray.kt:7]: I'm not sure about the naming, but `ByteArray` is a reserved name 😞

Comment (THenry14) [reply lib/src/main/kotlin/com/swmansion/starknet/data/types/StarknetByteArray.kt:7]: Yeah, a pretty common problem. StarknetByteArray seems reasonable :+1:

File: lib/src/main/kotlin/com/swmansion/starknet/data/types/StarknetByteArray.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    val data: List<Felt>,
+    val pendingWord: Felt,
+    val pendingWordLen: Int,
+) : ConvertibleToCalldata {
+    override fun toCalldata(): List<Felt> {
+        return listOf(data.size.toFelt) + data + listOf(pendingWord, pendingWordLen.toFelt)
```
Comment (THenry14) [inline lib/src/main/kotlin/com/swmansion/starknet/data/types/StarknetByteArray.kt:13]: can you give a bit more context in here? what is a pending word? and shouldn't the length be the first element (before pendingWord)? (maybe I just misunderstand/do not know something, hence my question)

Comment (DelevoXDG) [reply lib/src/main/kotlin/com/swmansion/starknet/data/types/StarknetByteArray.kt:13]: You can check out explanation of `ByteArray` fields and its serialization process in the [docs](https://docs.starknet.io/documentation/architecture_and_concepts/Smart_Contracts/serialization_of_Cairo_types/#serialization_of_byte_arrays).

Should we add a docstring that mentions that this class corresponds to `ByteArray` struct in Cairo, so there's no confusion about its purpose?

Comment (THenry14) [reply lib/src/main/kotlin/com/swmansion/starknet/data/types/StarknetByteArray.kt:13]: yes, definitely let's add this

**Review by THenry14 — APPROVED** [review]
can be merged after resolving all comments

**Review by THenry14 — APPROVED** [review]

**Pull request merged**